### PR TITLE
feat(next): add stripe sdk and csp

### PR DESCRIPTION
**Pull request by StaberindeZA**
## Because

- We want to start adding Stripe client elements to the checkout page.

## This pull request

- Adds Stripe SDK to a client component, which is added to Checkout.
- Adds middleware to add CSP to page requests.

## Issue that this pull request solves

Closes: #FXA-8847

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: libs/payments/ui/src/lib/client/components/StripeWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    currency,
+    paymentMethodCreation: 'manual',
+  };
+
+  return (
+    <div>
+      <Elements stripe={stripePromise} options={options}>
+        Hello there from Stripe
+      </Elements>
+    </div>
```
Comment (xlisachan) [inline libs/payments/ui/src/lib/client/components/StripeWrapper.tsx:28]: `div` can be removed.

File: libs/payments/ui/src/lib/client/components/StripeWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  currency: string;
+}
+
+export function StripeWrapper({ amount, currency }: StripeWrapperProps) {
+  // TODO - Load from config
+  const stripePromise = loadStripe(
+    'pk_test_VNpCidC0a2TJJB3wqXq7drhN00sF8r9mhs'
+  );
+
+  const options: StripeElementsOptions = {
```
Comment (xlisachan) [inline libs/payments/ui/src/lib/client/components/StripeWrapper.tsx:16]: Do we need to include locale in `options`?

File: libs/payments/cart/src/lib/cart.service.ts [hunk 3]
```diff
@@ -59,6 +56,9 @@ export class CartService {
       accountCustomer?.stripeCustomerId
     );
 
+    console.log('Eligiblity here');
+    console.log(eligibility);
+
```
Comment (xlisachan) [inline libs/payments/cart/src/lib/cart.service.ts:61]: :eyes:

File: apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/start/page.tsx [hunk 4]
```diff
@@ -158,6 +159,7 @@ export default async function Checkout({ params }: { params: CheckoutParams }) {
             </p>
           </div>
         </section>
+        <StripeWrapper amount={1099} currency="usd" />
```
Comment (xlisachan) [inline apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/start/page.tsx:162]: Should currency be hardcoded? Just temporarily?

File: libs/payments/ui/src/lib/client/components/StripeWrapper.tsx (hunk 2 above)
Comment (StaberindeZA) [reply libs/payments/ui/src/lib/client/components/StripeWrapper.tsx:16]: For now the default value, `auto`, uses the browser language, which matches our current app logic.

File: apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/start/page.tsx (hunk 4 above)
Comment (StaberindeZA) [reply apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/start/page.tsx:162]: Yup, both are temporary. These will change to real values once we have the actual invoice data as part of the cart data.

File: apps/payments/next/middleware.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import { NextRequest, NextResponse } from 'next/server';
```
Comment (pdehaan) [inline apps/payments/next/middleware.ts:1]:
```suggestion
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import { NextRequest, NextResponse } from 'next/server';
```

File: apps/payments/next/middleware.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+export function middleware(request: NextRequest) {
```
Comment (StaberindeZA) [inline apps/payments/next/middleware.ts:3]: This follows the example provided by Next.js docs, in addition to the Stripe CSPs

https://nextjs.org/docs/app/building-your-application/configuring/content-security-policy

File: libs/payments/ui/src/lib/client/components/StripeWrapper.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
+'use client';
```
Comment (pdehaan) [inline libs/payments/ui/src/lib/client/components/StripeWrapper.tsx:1]:
```suggestion
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

'use client';
```

**Review by xlisachan — APPROVED** [review]
LGTM, just need to add the copyright headers that @pdehaan mentioned! Thanks Reino!

**Pull request merged**